client/src/components/events-components/EventModal.tsx

```diff
@@ -0,0 +1,63 @@
+import type { EventWithAllData } from "@/utils/interfaces";
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogHeader,
+  DialogTitle,
+} from "../ui/dialog";
+import { useEffect, useState } from "react";
+import { getEvent } from "@/utils/eventService";
+
+interface EventModalProps {
+  open: boolean;
+  onOpenChange: (open: boolean) => void;
+  eventId: number;
+}
+const EventModal = ({ open, onOpenChange, eventId }: EventModalProps) => {
+  const [isEventLoading, setIsEventLoading] = useState(true);
+  const [event, setEvent] = useState<EventWithAllData | null>(null);
+  useEffect(() => {
+    const fetchEvent = async () => {
+      const fetchedEvent = await getEvent(eventId);
+      setEvent(fetchedEvent);
+      setIsEventLoading(false);
+    };
+    fetchEvent();
+  }, []);
```

Comment:
again, please consider limiting the usage of `useEffect` if you can, in this case, I think it would be better to fetch the event in the root, and pass `Event` directly into this component